ui/app/components/allocation-row.js

```diff
@@ -54,12 +56,15 @@ export default Component.extend({
 
   fetchStats: task(function*() {
     do {
-      try {
-        yield this.get('stats.poll').perform();
-        this.set('statsError', false);
-      } catch (error) {
-        this.set('statsError', true);
+      if (this.get('stats')) {
+        try {
+          yield this.get('stats.poll').perform();
+          this.set('statsError', false);
+        } catch (error) {
+          this.set('statsError', true);
+        }
       }
+
       yield timeout(500);
     } while (this.get('enablePolling'));
   }).drop(),
```

Comment:
> We have a task to fetch the stats that then runs another task to poll the stats.

Yes, that's correct. In order to prevent coordination, the stats tracker has its internal polling mechanism in addition to the call site polling mechanisms like you see here. This way the stats-tracker itself doesn't control its own state, but when there are multiple consumers of a single stats-tracker on a page at once (like here, with mem and cpu), they don't need to know about each other. [Read more about it here](https://github.com/hashicorp/nomad/pull/4704).

> If the poll task throws an error set a property to true, waits 500 milliseconds and then immediately try again?

Also correct. Basically we want to inform users that the stats are stale, but we don't want to give up because the error could be temporary.

> What happens if the job or allocation state changes or it disappears? Is that possible?

An allocation state can change. From pending to running, the chart will show up due to `stats` no longer being undefined (it's dependent on `allocation.isRunning`). From running to complete/failed/lost, the chart will disappear due to `stats` no longer being defined.

Reply:
Cool, thanks for the explanation! I'll have a quick read in the morning and shout if I have an other bits.